src/i18n/i18n-context.tsx

```diff
@@ -0,0 +1,36 @@
+import { useLocation } from "@solidjs/router";
+import { type JSX, createContext, useContext } from "solid-js";
+import { createStore } from "solid-js/store";
+import { createTranslator } from "./translator";
+import { getValidLocaleFromPathname } from "./helpers";
+
+type ProviderProps = { children: JSX.Element };
+
+const initialI18nStore = {
+	get t() {
+		const { pathname } = useLocation();
+		const locale = getValidLocaleFromPathname(pathname);
+
+		return createTranslator(locale);
+	},
+};
+
+export const I18nContext =
+	createContext<typeof initialI18nStore>(initialI18nStore);
+
+export function I18nProvider(props: ProviderProps) {
+	const [i18n] = createStore(initialI18nStore);
+
+	return (
+		<I18nContext.Provider value={i18n}>{props.children}</I18nContext.Provider>
+	);
+}
+
+export function useI18n() {
+	const context = useContext(I18nContext);
+	console.log(context);
```

Comment:
Leftover debug?

Reply:
correct!! 😅

taking it out in 3...2...